[MFTF] 1006 - MSI MFTF Test Cases - Part 3

ivy00johns opened the pull request:
### Description
This PR contains 10 additional tests for MSI along with new Suite/Metadata/Data/Action Groups for MSI Tests to help speed up the execution of Tests.

### Suite
- dev/tests/acceptance/tests/_suite/msi-suite.xml

### Tests
- [Configuration-Sources] Filling Source Code with space on Source Creation Page not allowed: https://app.zenhub.com/workspace/o/magento-engcom/msi/issues/939
- [Configuration-Multiple Sources] For each source Qty can be set by Admin user: https://app.zenhub.com/workspace/o/magento-engcom/msi/issues/861
- [Configuration-Multiple Sources] Sources can be removed by clicking delete icon for each source by Admin user on Product page: https://app.zenhub.com/workspace/o/magento-engcom/msi/issues/937
- [Configuration-Sources] Admin user applied Save and load data views in Manage Sources grid: https://app.zenhub.com/workspace/o/magento-engcom/msi/issues/915
- [Configuration-Stocks] Admin user applied Save and load data views in Manage Stocks grid: https://app.zenhub.com/workspace/o/magento-engcom/msi/issues/916
- [Configuration-Stocks-Delete Stock] Stock deleted from Stock page by Admin user when it has no Sales Channel assigned: https://app.zenhub.com/workspace/o/magento-engcom/msi/issues/949
- [Configuration-Stocks] Sources assigned to the Stock can be deleted by clicking trash icon by Admin user: https://app.zenhub.com/workspace/o/magento-engcom/msi/issues/951
- [Configuration-Catalog-Products-Simple Product] Order created with Simple product from Admin: Admin user: https://app.zenhub.com/workspace/o/magento-engcom/msi/issues/1008
- [Catalog-Products-Virtual Product] Virtual Product assigned to Test Stock created by Admin user: Admin user: https://app.zenhub.com/workspace/o/magento-engcom/msi/issues/1042
- [Sales-Orders-Orders with Virtual Product] Guest Customer ordered Virtual product on Test source from Homepage: https://app.zenhub.com/workspace/o/magento-engcom/msi/issues/1043

### MetaData/Data
- stock-source-link-meta.xml
- MsiSalesChannelData.xml
- MsiSourceViewData.xml
- MsiStockExtensionAttributeData.xml
- MsiStockViewData.xml
- MsiStoreViewData.xml
- MsiWebsiteData.xml
- StockSourceLinksData.xml

### ActionGroups
- AdminGoToProductGridAddNewProductByTypeActionGroup.xml
- AdminGoToProductGridFilterResultsByInputActionGroup.xml
- AdminGoToSourceGridFilterResultsByInputActionGroup.xml
- AdminGoToStockGridFilterResultsByInputActionGroup.xml
- AdminGoToStockPageAssignStockToSourceAndMainWebsiteActionGroup.xml
- AdminOnProductEditPageAssignSourceToProductActionGroup.xml
- AdminOnStockEditPageAssignSourceToStockActionGroup.xml

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

ivy00johns committed:
magento-engcom/msi#1006 [MFTF] Automated MSI Test Cases - Part 3

- Adding additional elements to the Source Section.
- Adding new Source/Stock Tests.
- Adding Source/Stock data for the new Tests.

ivy00johns committed:
magento-engcom/msi#1006 [MFTF] Automated MSI Test Cases - Part 3

- Adding an additional element to the Source Section.
- Adding a Delete Sources assigned to Stock Test.

ivy00johns committed:
magento-engcom/msi#1006 [MFTF] Automated MSI Test Cases - Part 3

- Adding conditionalClicks to the "AddRemove_ColumnsTest"'s to prevent random failures.
- Adding a Simple Product Checkout via the Admin with Stock/Sources assigned Test.
- Adding MSI Customer data for the Checkout Test.

ivy00johns committed:
magento-engcom/msi#1006 [MFTF] Automated MSI Test Cases - Part 3

- Removing "skip" group from a Test. The bug was resolved.
- Adding Steps to assign the Stock to the Source that was created via the API.
- Adding Steps to assign the Source that was created via the API.
- Adding a Conditional Click step for the Checkout test. IF the "Select Store View" page is present click on the correct Store View.
- Shortening the Website data names so they don't break the page.

ivy00johns committed:
magento-engcom/msi#1006 [MFTF] Automated MSI Test Cases - Part 3

- Merging 2.3-develop

ivy00johns committed:
magento-engcom/msi#1006 [MFTF] Automated MSI Test Cases - Part 3

- Adding the Virtual Product assigned to a Stock test.
- Adding data for a Virtual Product.

ivy00johns committed:
magento-engcom/msi#1006 [MFTF] Automated MSI Test Cases - Part 3

- Adding a missing <waitForPageLoad>.
- Correcting a Save button selector.
- Un-skipping 2 tests skipped due to a bug that has been resolved.

ivy00johns committed:
magento-engcom/msi#1006 [MFTF] Automated MSI Test Cases - Part 3

- Adding new Test for Create Order with a Virtual Product.

ivy00johns committed:
magento-engcom/msi#1006 [MFTF] Automated MSI Test Cases - Part 3

- Adding a new Action Group for AdminGoToStockPageAssignStockToSourceAndMainWebsite.
- Updating Tests to use the new Action Group.
- Expanding the Stock Data that is used in tests. Including website.

ivy00johns committed:
magento-engcom/msi#1006 [MFTF] Automated MSI Test Cases - Part 3

- Adding new Action Groups for Stock/Source Grid pages.
- Updating Tests to use the new Action Groups.

ivy00johns committed:
magento-engcom/msi#1006 [MFTF] Automated MSI Test Cases - Part 3

- Updating Tests to use the new Action Groups.
- Un-skipping test due to bug resolution.

ivy00johns committed:
magento-engcom/msi#1006 [MFTF] Automated MSI Test Cases - Part 3

- Adding additional metadata for linking Sources to Stocks.
- Adding additional metadata for linking Websites to Stocks.
- Adding additional data for linking Sources to Stocks.
- Adding additional data for linking Websites to Stocks.
- Updating tests to use the new metadata for linking Websites/Sources to Stocks.
- Adding clean up steps to tests.

ivy00johns committed:
magento-engcom/msi#1006 [MFTF] Automated MSI Test Cases - Part 3

- Merging 2.3-develop

okolesnyk reviewed (changes requested):
@ivy00johns please fetch latest mainline from MSI 2.3-develop branch and move tests from Acceptance to MFTF directory.

ivy00johns committed:
magento-engcom/msi#1006 [MFTF] Automated MSI Test Cases - Part 3

- Merging 2.3-develop to my branch.

ivy00johns committed:
magento-engcom/msi#1006 [MFTF] Automated MSI Test Cases - Part 3

- Renaming the "Acceptance" folder to "MFTF" so it works with the latest version of MFTF.

ivy00johns committed:
magento-engcom/msi#1006 [MFTF] Automated MSI Test Cases - Part 3

- Fixing an incorrect path for the Action Group schema.
- Correcting a few invalid data references in Action Groups.
- Updating Tests to use non-default Stocks unless they are explicitly Testing the "Default Stock".

ivy00johns committed:
magento-engcom/msi#1006 [MFTF] Automated MSI Test Cases - Part 3

- Introducing Suites for "Single Stock Mode" and "Multi Stock Mode".
- Adding "single_stock_mode" groups to Tests that need to be in "Single Stock Mode" to pass.
- Removing Source creation from the "Single Stock Mode" tests.
- Updating Tests so they are configured for "Single Stock Mode".

ivy00johns committed:
magento-engcom/msi#1006 [MFTF] Automated MSI Test Cases - Part 3

- Adding "multi_mode" group for Tests that need to be run in the "Multi Source/Stock Modes".
- Updating "single_stock_mode" to "single_mode".
- Updating Suite names and group tags.

ivy00johns committed:
magento-engcom/msi#1006 [MFTF] Automated MSI Test Cases - Part 3

- Updating "<stories>" tags in all Tests for better organization in the Allure report.

magento-cicd2 commented:
[![CLA assistant check](https://cla.dev.magento.com/pull/badge/signed)](https://cla.dev.magento.com/magento-engcom/msi?pullRequest=1092) <br/>All committers have signed the CLA.

ivy00johns commented:
Changes implemented.
- Renamed "Acceptance" to "MFTF".
- Updated the "stories" tag in each Test to make them more useful in the Allure report, per Sheila's request.
- Added a .suite file for "single_mode" (Single [Default] Stock/Source) and "multi_mode" (Multiple Stocks/Sources) since there are 2 general modes for MSI.
- Added a new group to the Tests. One for "single_mode" and "multi_mode".

okolesnyk committed:
Merge branch '2.3-develop' of github.com:magento-engcom/msi into 1006-MSI-MFTF-Test-Cases-3

okolesnyk commented:
Pull Request closed because branch has inappropriate merges.
Here is new PR replaces this one: https://github.com/magento-engcom/msi/pull/1238
